### PR TITLE
Refactor server

### PR DESCRIPTION
**Pull request by pnuu**
Another refactoring PR, now for the Server code.

### CODE REVIEW

File: trollmoves/server.py [hunk 1]
```diff
@@ -418,32 +463,29 @@ def process_notify(orig_pathname, publisher, pattern, attrs):
...
-    elif (os.stat(orig_pathname).st_size == 0):
-        # Want to avoid files with size 0.
+    elif os.stat(orig_pathname).st_size == 0:
         LOGGER.debug("Ignoring empty file: %s", orig_pathname)
         return
     else:
         LOGGER.debug('We have a match: %s', orig_pathname)
 
     pathname = unpack(orig_pathname, **attrs)
+    info = _get_inotify_message_info(attrs, orig_pathname, pathname)
```
Comment (mraspaud) [inline trollmoves/server.py:473]: I'm a bit confused by the function name here, but maybe I'm not really following: what does inotify have to do in the posttroll notifier function?

**Review by mraspaud — COMMENTED** [review]
Thanks a lot for the huge refactoring work! Just one clarification needed for my part

File: trollmoves/server.py (hunk 1 above)
Comment (pnuu) [reply trollmoves/server.py:473]: Umm, the `_get_inotify_message_info()` is called in `process_notify()` function which is called only in the nested function `process_notify_publish()` of `create_inotify_notifier()`. So it is part of the inotify chain :-) I'll adjust the ordering of the functions to make this more clear.

Comment (pnuu) [reply trollmoves/server.py:473]: Oh, you were right. It is used as a callback (sigh) in all the `create_*_notifier()` functions.

File: trollmoves/server.py [hunk 2]
```diff
@@ -192,7 +213,7 @@ def push(self, message):
         new_msg = self._move_files(message)
         if new_msg is None:
             new_msg = Message(message.subject,
-                              message.type,
+                              'file',
```
Comment (mraspaud) [inline trollmoves/server.py:216]: Don't we have `dataset` types of messages coming here?

**Review by mraspaud — COMMENTED** [review]
One question, otherwise LGTM

File: trollmoves/server.py (hunk 2 above)
Comment (pnuu) [reply trollmoves/server.py:216]: Could be? I don't remember seeing those in my own setups. If you have such a message, I could perhaps adapt this. Or would it be sufficient to check the message data for `'uri'`, `'dataset'` and `'collection'` and decide based on that? The incomming message in any case has type `'push'`.

Comment (pnuu) [reply trollmoves/server.py:216]: Oh, I guess @TAlonglong had one example in an issue, I'll have a look after lunch.

Comment (pnuu) [reply trollmoves/server.py]: Ok, done.

**Pull request merged**